packages/dashboard/lib/DashboardServer.ts

```diff
@@ -95,6 +98,62 @@ export class DashboardServer {
       this.expressApp.post("/rpc", this.postRpc.bind(this));
     }
 
+    this.expressApp.get("/fetch-and-compile", async (req, res) => {
+      const { address, networkId, etherscanApiKey } = req.query as Record<
+        string,
+        string
+      >;
+      let config;
+      try {
+        config = Config.detect();
+      } catch (error) {
+        const notFound = "Could not find suitable configuration file.";
+        if (!error.message.includes(notFound)) {
+          throw error;
+        }
+      }
+
+      // a key provided in the browser takes precedence over on in the config
+      let etherscanKey: undefined | string;
+      if (etherscanApiKey) {
+        etherscanKey = etherscanApiKey;
+      } else if (config && config.etherscan !== undefined) {
+        etherscanKey = config.etherscan.apiKey;
+      }
+
+      config = Config.default().merge({
+        networks: {
+          custom: { network_id: networkId }
+        },
+        network: "custom",
+        etherscan: {
+          apiKey: etherscanKey
+        }
+      });
+
+      let result;
+      try {
+        result = (await fetchAndCompile(address, config)).compileResult;
+      } catch (error) {
+        if (!error.message.includes("No verified sources")) {
+          throw error;
+        }
+      }
```

Comment:
Also don't throw if no sources? It can be handled like how `!result` is handled below with `res.json({ compilations: [] })`

Reply:
The point of this is that `fetchAndCompile` throws with that error message in the case that it can't find verified sources. We want to not throw in that case, but rather throw when it fails in some other unexpected way. Does that make sense?

Reply:
Yes. I was just thinking it'd be less work to write a case to receive a 500 in react? Maybe we can `console.error` it, which should show in the Dashboard terminal, and still send the empty compilations response?